common.go

```diff
@@ -29,30 +30,34 @@ var AllowTrace = false
 
 // Param - Get a url parameter by name
 func Param(r *http.Request, name string) string {
-	return r.URL.Query().Get(":" + name)
+	// use the request context
+	if v, ok := r.Context().Value("vestigo_" + name).(string); ok {
+		return v
+	}
+	return ""
 }
 
 // ParamNames - Get a url parameter name list
 func ParamNames(r *http.Request) []string {
-	var names []string
-	for k := range r.URL.Query() {
-		if strings.HasPrefix(k, ":") {
-			names = append(names, k)
-		}
+	if v, ok := r.Context().Value("vestigo_param_names").([]string); ok {
+		return v
 	}
-	return names
+	return []string{}
 }
 
 // AddParam - Add a vestigo-style parameter to the request -- useful for middleware
 // Appends :name=value onto a blank request query string or appends &:name=value
 // onto a non-blank request query string
 func AddParam(r *http.Request, name, value string) {
-	q := url.QueryEscape(":"+name) + "=" + url.QueryEscape(value)
-	if r.URL.RawQuery != "" {
-		r.URL.RawQuery += "&" + q
-	} else {
-		r.URL.RawQuery += q
+	paramNames := []string{name}
+	if v, ok := r.Context().Value("vestigo_param_names").([]string); ok {
```

Comment:
Do not use a plain string type like that. See https://blog.golang.org/context#TOC_3.2.